doc/manuals/media-files.rst

```diff
@@ -38,6 +38,50 @@ The image URL will have a checksum embedded in it so that when the
 contents of the media class is changed, all images which use that
 media class will be regenerated to reflect the new media class.
 
+Responsive images
+.................
+
+To provide images in multiple `responsive sizes`_, use the ``srcset`` attribute::
+
+    %% templates/mediaclass.config
+
+    [
+        {"masthead", [
+            {width, 1600},
+            {height, 900},
+            {crop, center},
+            upscale,
+            {quality, 85},
+            {srcset, [
+                {"640w", [{quality, 50}]},
+                {"1200w", []},
+                {"2x", []}
+            ]},
+            {sizes, "100vw"}
+        ]}
+    ].
+
+An ``{% image id mediaclass="masthead" %}`` tag in your template will output::
+
+    <img src='image-default.jpg'
+        sizes='100vw'
+        srcset='image-640.jpg 640w, image-1200.jpg 1200w, image-2400.jpg 2x' />
+
+Each ``srcset`` value is either a `width descriptor`_ or a pixel density
+descriptor.
+
+* A width descriptor of ``640w`` will resize the image to a width of 640 pixels.
+* A pixel density descriptor of ``2x`` will resize the image to 2 times the
+  original media class width value, so 2 * 1600 = 3200.
+
+By default, each srcset image will inherit all properties from the parent
+media class. So, in the example above, the 640w image will have a reduced
```

Comment:
You would need a bare parent that only implements the common image attributes.
To make this flexible it would be useful to reference other configs:

```
{"masthead", [
    {width, 1600},
    {height, 900},
    {srcset, [
        {"640w", "masthead-small"},
        {"1200w", "masthead-large"},
        {"2x", []}
    ]}
]},
{"masthead-small", [
    {crop, center},
    {quality, 85}
]},
{"masthead-large", [
    {quality, 50}
]}

```